tratto/src/main/java/star/tratto/util/javaparser/JDoctorUtils.java

```diff
@@ -0,0 +1,210 @@
+    }
+
+    public static String convertConditionParameterType(String conditionParameterType) {
+        List<String> primitiveConditionsValues = ConditionPrimitiveType.getConditionValues();
+        conditionParameterType = removeSpuriousCharacters(conditionParameterType);
+        if (primitiveConditionsValues.contains(conditionParameterType.replaceAll("[^a-zA-Z]+", ""))) {
+            conditionParameterType = convertToPrimitiveType(conditionParameterType);
+        }
+        if (conditionParameterType.startsWith("[")) {
+            conditionParameterType = convertConditionToArrayType(conditionParameterType);
+        }
+        List<String> paramPathList = getPathList(conditionParameterType);
+        return getClassNameFromPathList(paramPathList);
+    }
+
+    public static String convertToPrimitiveType(String primitiveType) {
+        List<String> primitiveConditionsValues = ConditionPrimitiveType.getConditionValues();
+        if (primitiveConditionsValues.contains(primitiveType.replaceAll("[^a-zA-Z]+", ""))) {
+            String conditionPrimitiveRegex = ConditionPrimitiveType.getConditionRegexValues();
+            String regex = String.format(
+                    "[^A-Za-z0-9_]*(%s)[^A-Za-z0-9_]*",
+                    conditionPrimitiveRegex
+            );
+
+            Pattern pattern = Pattern.compile(regex);
+            Matcher matcher = pattern.matcher(primitiveType);
+
+            if (matcher.find()) {
+                String extractedPrimitiveType = matcher.group(1);
+                ConditionPrimitiveType conditionParamEnum = ConditionPrimitiveType.convertValue(extractedPrimitiveType);
+                ConditionPrimitiveType conditionParamConverted = ConditionPrimitiveType.condition2jp(conditionParamEnum);
+                return primitiveType.replaceAll(conditionPrimitiveRegex, conditionParamConverted.getValue());
+            } else {
+                String errMsg = String.format(
+                        "The condition parameter does not match any primitive type: %s",
+                        primitiveType
+                );
+                throw new IllegalArgumentException(errMsg);
+            }
+        }
+        String errMsg = String.format("The string passed to the function (%s) does not represent a primitive type", primitiveType);
+        throw new IllegalArgumentException(errMsg);
+    }
+
+    public static String getClassNameFromPathList(
+            List<String> pathList
+    ) {
+        return pathList.get(pathList.size() - 1);
+    }
+
+    public static String getExtendsType(String jpMethodString, String jpParamString) {
+        // count dimension of array
+        int howMany = 0;
+        for (char c : jpParamString.toCharArray()) {
+            if (c == '[') {
+                howMany++;
+            }
+        }
+        String regex = String.format("%s\\s+extends\\s+([A-Za-z0-9_]+)[<[A-Za-z0-9_,]+]*", jpParamString.replaceAll("\\[\\]",""));
```

Comment:
Can you pass around data structures rather than strings, to avoid the need for this regular expression?